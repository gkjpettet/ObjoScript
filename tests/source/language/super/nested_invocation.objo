class Person {
greet(who) {
System.print("Hello " + who + " from Person")
}
}

class Doctor is Person {
greet(who) {
System.print("Hello " + who + " from Doctor")
super.greet(who)
}
}

class Radiologist is Doctor {
greet(who) {
System.print("Hello " + who + " from Radiologist")
super.greet(who)
}
}

var aoife = Person()
aoife.greet("stranger") # Expect: Hello stranger from Person

var fi = Doctor()
fi.greet("stranger")
# Expect: 
# Hello stranger from Doctor
# Hello stranger from Person

var garry = Radiologist()
garry.greet("stranger")
# Expect:
# Hello stranger from Radiologist
# Hello stranger from Doctor
# Hello stranger from Person