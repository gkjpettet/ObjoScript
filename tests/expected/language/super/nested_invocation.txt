Hello stranger from Person
Hello stranger from Doctor
Hello stranger from Person
Hello stranger from Radiologist
Hello stranger from Doctor
Hello stranger from Person